webapp/src/Service/ImportProblemService.php

```diff
@@ -793,4 +796,67 @@ public function importZippedProblem(
 
         return $problem;
     }
+
+    public function importProblemFromRequest(Request $request, ?int $contestId = null): array
+    {
+        $file = $request->files->get('zip');
+        if (empty($file)) {
+            throw new BadRequestHttpException('ZIP file missing');
+        }
+
+        $contest = null;
+        if ($contestId) {
+            /** @var Contest $contest */
+            $contest = $this->em->getRepository(Contest::class)->find($contestId);
+        }
+        $allMessages = [];
+
+        // Only timeout after 2 minutes, since importing may take a while.
+        set_time_limit(120);
```

Comment:
Do we really have problems which need more than 60 seconds?

Reply:
For sure